Pensel/firmware/modules/utilities/scheduler.h

```diff
@@ -0,0 +1,32 @@
+/*!
+ * @file    scheduler.h
+ * @author  Tyler Holmes
+ * @version 0.1.0
+ * @date    28-April-2018
+ * @brief   A rudamentary scheduler
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include "common.h"
+
+#define NUM_SLOTS (16)
+#define SCHEDULER_FINISHED  (-1)
+
+typedef struct {
+    uint8_t num_slots_used;
+    ret_t (*callbacks[NUM_SLOTS])(int32_t * callback_time_ms);
+    int32_t callback_times_ms[NUM_SLOTS];
+    int32_t callback_last_run_ms[NUM_SLOTS];
+} schedule_t;
```

Comment:
Thoughts on creating a single `callback_t` struct that will hold the callback and time stamps? This struct could then simply the last three elements to a single `callback_t callbacks[NUM_SLOTS]` entry.

Reply:
I like it!